daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Validation.scala

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
```

Comment:
this seems like it is somewhat overlapping with #7824 so unless that is incorrect, I’ll review this once we’ve merged the other one.